lib/grape/middleware/oink_middleware.rb

```diff
@@ -3,13 +3,13 @@ module Middleware
     class OinkMiddleware < Grape::Middleware::Base
       def before
         options = env['api.endpoint'].try(:options)
-        
+
         if options.present?
           controller = options[:for].to_s.split("::").last.underscore
           action = options[:path].try(:first)
           if action
-            action.gsub!("/", "sl_")
-            action.gsub!(":", "col_")
+            action.to_s.gsub!("/", "sl_")
```

Comment:
Nice catch. One issue is that `action.to_s` creates a new string copy each time, so this wouldn't apply to `action`. This is easily fixable though, similar to the below.

```ruby
action = action.to_s
action.gsub!("/", "sl_")
action.gsub!(":", "col_")
```